os/fs/vfs/fs_poll.c

```diff
@@ -273,36 +273,41 @@ int file_poll(FAR struct file *filep, FAR struct pollfd *fds, bool setup)
 	FAR struct inode *inode;
 	int ret = -ENOSYS;
 
-	DEBUGASSERT(filep != NULL && filep->f_inode != NULL);
+	DEBUGASSERT(filep != NULL);
```

Comment:
I understand the idea
Calling select on a closed file may be a mistake from the app side, we do not need to have an assert failure.
We can just report the error.

But can you help me understand how and why the need to make this change came up?

Reply:
To prevent unexpected error in product level.